File: src/frontend/src/app/components/atendimento/convenio/model/convenio-dto.ts
import { Exclude, Expose } from 'class-transformer';
import { CodigoConvenioDTO } from './codigo-convenio-dto';
import { ConvenioTipoCobranca } from './convenio-tipo-cobranca.enum';

@Exclude()
export class ConvenioDTO {
  @Expose() id?: string;
  @Expose() nome?: string;
  @Expose() pessoaId?: string;
  @Expose() pessoaNome?: string;
  @Expose() registroAns?: string;
  @Expose() ativo?: boolean;
  @Expose() tipoCobranca?: ConvenioTipoCobranca;
  @Expose() codigos?: CodigoConvenioDTO[];
  @Expose() createdAt?: Date;
  @Expose() updatedAt?: Date;
  @Expose() createdBy?: string;
  @Expose() updatedBy?: string;
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/atendimento/convenio/dto/ConvenioDTO.java
package br.com.grupopipa.gestaointegrada.atendimento.convenio.dto;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.atendimento.codigoconvenio.dto.CodigoConvenioDTO;
import br.com.grupopipa.gestaointegrada.atendimento.convenio.entity.ConvenioTipoCobrancaEnum;
import br.com.grupopipa.gestaointegrada.core.dto.DTO;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class ConvenioDTO implements DTO {

    private UUID id;
    private String nome;

    private UUID pessoaId;
    private String pessoaNome;

    private String registroAns;
    private Boolean ativo;
    private ConvenioTipoCobrancaEnum tipoCobranca;

    private List<CodigoConvenioDTO> codigos;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String createdBy;
    private String updatedBy;
}
